index.js

```diff
@@ -417,6 +419,33 @@ Ycb.prototype = {
                 }
             }
         }
+
+        this._buildUsageMap(dimsUsed);
+    },
+
+    _buildUsageMap: function (dimsUsed) {
+        var i, j, k,
+            value,
+            name,
+            hierarchy;
+
+        // Assemble map of all dimensions used including ancestry
+        this.dimsUsed = {};
+        for (name in dimsUsed) {
+            if (dimsUsed.hasOwnProperty(name) && this._dimensionHeierarchies.hasOwnProperty(name)) {
+                this.dimsUsed[name] = {};
+                for (i=0; i<dimsUsed[name].length; i += 1) {
+                    value = dimsUsed[name][i];
+                    for (hierarchy in this._dimensionHeierarchies[name]) {
+                        if (this._dimensionHeierarchies[name].hasOwnProperty(hierarchy)) {
+                            if (-1 !== this._dimensionHeierarchies[name][hierarchy].indexOf(value)) {
```

Comment:
This also marks any parents of a referenced dimension to also be marked as used. This allows constant-time lookups in `_makeOrderedLookupList`.